compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -4039,6 +4084,47 @@ abstract class GenJSCode extends plugins.PluginComponent
           js.Undefined()
         else
           genPrimitiveJSClass(classSym)
+      } else if (code == CREATE_INNER_JS_CLASS || code == CREATE_LOCAL_JS_CLASS) {
+        val classSym = resolveReifiedJSClassSym(args(0))
+        val superClassValue = genExpr(args(1))
+        if (classSym == NoSymbol) {
+          js.Undefined()
+        } else {
+          val captureValues = {
+            if (code == CREATE_INNER_JS_CLASS) {
+              val outer = genThis()
+              List.fill(classSym.info.decls.count(_.isClassConstructor))(outer)
```

Comment:
Doesn't this make adding a constructor to a public nested class a binary incompatible change? That would be a problem, no?

Reply:
No, it doesn't, because calls to `createInnerJSClass` are always in the same compilation unit as the inner class itself, so they will always be in sync. If you add a constructor to that inner class, anyway you're recompiling the enclosing class, which will therefore receive a new call `createInnerJSClass`.